devito/equation.py

```diff
@@ -89,4 +91,7 @@ def solve(eq, target, **kwargs):
     # turnaround time
     kwargs['rational'] = False  # Avoid float indices
     kwargs['simplify'] = False  # Do not attempt premature optimisation
+    if hasattr(eq, 'lhs'):
```

Comment:
what? recently I've seen this ``hasattr`` too often.

can you not just ``if eq.is_Equality`` ? and btw, what else should/could it be?

Reply:
>and btw, what else should/could it be?
it can be a single expression as solve defaults to `Eq(expr, 0)` if called as `solve(expr)` in which case `expr` does not have `is_Equality` attribute

Reply:
why not? all sympy expressions, even symbols, have ``is_Equality``